Flaky test in CI

failures:

---- exporter::config::collector::http_client::collector_client_tests::test_bring_your_own_client stdout ----
Error: ExportFailed(ConfigError { pipeline_name: "collector", config_name: "collector_endpoint", reason: "invalid uri from environment variable, invalid uri character" })

---- exporter::config::collector::tests::test_collector_exporter stdout ----
thread 'exporter::config::collector::tests::test_collector_exporter' panicked at opentelemetry-jaeger/src/exporter/config/collector/mod.rs:580:9:
assertion failed: exporter.is_ok()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    exporter::config::collector::http_client::collector_client_tests::test_bring_your_own_client
    exporter::config::collector::tests::test_collector_exporter

test result: FAILED. 9 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.11s

error: test failed, to rerun pass `-p opentelemetry-jaeger --lib`
error: process didn't exit successfully: `/home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo test --tests --manifest-path /home/runner/work/opentelemetry-rust/opentelemetry-rust/Cargo.toml --target-dir /home/runner/work/opentelemetry-rust/opentelemetry-rust/target/llvm-cov-target --locked --all-features --workspace` (exit status: 101)

> The failures are coming from the opentelemetry-jaeger crate. I can take on removing these tests as the crate is on track for deprecation

Yes please. It should be okay to disable the tests asap, and then do a release as well, so we can actually proceed with removing it completely.


test exporter::config::agent::tests::set_socket_address ... ok

failures:

---- exporter::config::collector::http_client::collector_client_tests::test_bring_your_own_client stdout ----
Error: ExportFailed(ConfigError { pipeline_name: "collector", config_name: "collector_endpoint", reason: "invalid uri from environment variable, invalid uri character" })

---- exporter::config::collector::tests::test_collector_exporter stdout ----
thread 'exporter::config::collector::tests::test_collector_exporter' panicked at opentelemetry-jaeger/src/exporter/config/collector/mod.rs:580:9:
assertion failed: exporter.is_ok()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    exporter::config::collector::http_client::collector_client_tests::test_bring_your_own_client
    exporter::config::collector::tests::test_collector_exporter

test result: FAILED. 9 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.12s

error: test failed, to rerun pass `-p opentelemetry-jaeger --lib`
error: process didn't exit successfully: `/home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo test --tests --manifest-path /home/runner/work/opentelemetry-rust/opentelemetry-rust/Cargo.toml --target-dir /home/runner/work/opentelemetry-rust/opentelemetry-rust/target/llvm-cov-target --locked --all-features --workspace` (exit status: 101)
Error: Process completed with exit code 1.

--- a/opentelemetry-jaeger/src/exporter/config/collector/http_client.rs
+++ b/opentelemetry-jaeger/src/exporter/config/collector/http_client.rs
@@ -188,6 +188,7 @@ mod collector_client_tests {
     use opentelemetry::trace::TraceError;
     use opentelemetry_sdk::runtime::Tokio;
 
+    // Ignore this test as it is flaky and the opentelemetry-jaeger is on-track for deprecation
     #[ignore]
     #[test]
     fn test_bring_your_own_client() -> Result<(), TraceError> {

--- a/opentelemetry-jaeger/src/exporter/config/collector/http_client.rs
+++ b/opentelemetry-jaeger/src/exporter/config/collector/http_client.rs
@@ -188,6 +188,7 @@ mod collector_client_tests {
     use opentelemetry::trace::TraceError;
     use opentelemetry_sdk::runtime::Tokio;
 
+    #[ignore]
     #[test]
     fn test_bring_your_own_client() -> Result<(), TraceError> {
         let invalid_uri_builder = new_collector_pipeline()

--- a/opentelemetry-jaeger/src/exporter/config/collector/mod.rs
+++ b/opentelemetry-jaeger/src/exporter/config/collector/mod.rs
@@ -571,6 +571,7 @@ mod tests {
         assert!(valid_uri.is_ok());
     }
 
+    // Ignore this test as it is flaky and the opentelemetry-jaeger is on-track for deprecation
     #[ignore]
     #[test]
     fn test_collector_exporter() {

--- a/opentelemetry-jaeger/src/exporter/config/collector/mod.rs
+++ b/opentelemetry-jaeger/src/exporter/config/collector/mod.rs
@@ -571,6 +571,7 @@ mod tests {
         assert!(valid_uri.is_ok());
     }
 
+    #[ignore]
     #[test]
     fn test_collector_exporter() {
         let exporter = new_collector_pipeline()

